src/event.rs

```diff
@@ -1268,3 +1286,23 @@ pub enum MouseScrollDelta {
     /// and move the content right and down (to reveal more things left and up).
     PixelDelta(PhysicalPosition<f64>),
 }
+
+bitflags::bitflags! {
+    #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+    pub struct WindowState: u32 {
```

Comment:
I don't think a bit flag is appropriate here, since the states are mutually exclusive, are they not?

Perhaps an enum like this is better (names up for bikeshedding):
```rust
enum WindowFrameState {
    Fullscreen, // Possibly containing `Fullscreen`?
    Maximized,
    Minimized,
    Normal,
}
```

And given that we now have this enum, perhaps we can unify it with setting the window (frame) state? So e.g. `window
set_frame_state(WindowFrameState::Maximized)` instead of `window.set_maximized(true)`?

Reply:
The state might not be mutually exclusive in the future, at least If I expose more stuff from Wayland it won't be.

I don't think we should change the functions, because fullscreen is attached to the particular monitor. At least for now I'd keep it the way it is.

Reply:
> The state might not be mutually exclusive in the future, at least If I expose more stuff from Wayland it won't be.

Maybe we should figure this out when we get there, even on Wayland the current states are mutually exclusive, so maybe additional states for Wayland should be dealt with in a different way while allowing states that are mutually exclusive to stay that way.

Reply:
They are `tiled` states, saying that some edge of the window is being tiled, it's basically the same as maximized.

Reply:
I don't know what `tiled` is, but what does "basically the same" mean? Is it mutually exclusive or not?
At least on other backends it is mutually exclusive, right?

Reply:
They are not, you can have them attached to any state, you can have 4 of them at the same time, etc. 

Tiled means that your window is, well, tiled to some screen edge or just generally tiled.

Reply:
Wow, that's really strange.
Wouldn't this completely change the meaning of what's exposed here? Like it doesn't make the API really cross-platform because the meaning is so different.

In any case, if it's not mutually exclusive, we can leave it as is.